src/Radicle/Internal/Eval.hs

```diff
@@ -62,7 +62,10 @@ specialForms = Map.fromList $ first Ident <$>
               Vec atoms_ -> do
                 atoms <- traverse isAtom (toList atoms_) ?? toLangError (SpecialForm "fn" "One of the arguments was not an atom")
                 e <- gets bindingsEnv
-                pure (Lambda atoms (b :| bs) e)
+                pure (Lambda (PosArgs atoms) (b :| bs) e)
+              Atom name -> do
+                e <- gets bindingsEnv
+                pure (Lambda (VarArgs name) (b :| bs) e)
               _ -> throwErrorHere $ SpecialForm "fn" "First argument must be a vector of argument atoms"
```

Comment:
These two error messages need to be updated.

Reply:
Let me know if the new wording here is confusing.